clients/hadoopfs/src/main/java/io/lakefs/LakeFSFileSystem.java

```diff
@@ -81,11 +119,23 @@ public void initialize(URI name, Configuration conf) throws IOException {
      * regardless of the given file path.
      */
     @Override
-    public FSDataInputStream open(Path path, int i) throws IOException {
-        LOG.debug("$$$$$$$$$$$$$$$$$$$$$$$$$$$$ Calling open method for: {} $$$$$$$$$$$$$$$$$$$$$$$$$$$$", path.getName());
-        String strToWrite = "abc";
-        MyInputStream inputStream = new MyInputStream(strToWrite);
-        return new FSDataInputStream(inputStream);
+    public FSDataInputStream open(Path path, int bufSize) throws IOException {
+	try {
+	    LOG.debug("$$$$$$$$$$$$$$$$$$$$$$$$$$$$ open(" + path.getName() + ") $$$$$$$$$$$$$$$$$$$$$$$$$$$$");
+
+	    ObjectsApi objects = new ObjectsApi(apiClient);
+	    ParsedPath pp = parsePath(path);
+	    ObjectStats stats = objects.statObject(pp.repo, pp.ref, pp.path);
+	    URI physicalUri = translateUri(new URI(stats.getPhysicalAddress()));
+
+	    Path physicalPath = new Path(physicalUri.toString());
+	    FileSystem physicalFs = physicalPath.getFileSystem(conf);
+	    return physicalFs.open(physicalPath, bufSize);
+	} catch (io.lakefs.clients.api.ApiException e) {
+	    throw new RuntimeException("API exception: " + e.getResponseBody());
```

Comment:
think you can pass `e` as inner exception here

Reply:
Yup, forgot about exception causes (aka "Go `%w` done right"...).  Thanks!